admin_setting.go

```diff
@@ -0,0 +1,461 @@
+package tfe
+
+import (
+	"context"
+)
+
+// Compile-time proof of interface implementation.
+var _ AdminSettings = (*adminSettings)(nil)
+
+// AdminSettings describes all the admin settings related methods that the Terraform Enterprise API supports.
+// Note that admin settings are only available in Terraform Enterprise.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/admin/settings.html
+type AdminSettings interface {
+	// GetGeneral returns the general settings
+	GetGeneral(ctx context.Context) (*AdminGeneralSetting, error)
+
+	// UpdateGeneral updates general settings.
+	UpdateGeneral(ctx context.Context, options AdminGeneralSettingsUpdateOptions) (*AdminGeneralSetting, error)
+
+	// GetCostEstimation returns the cost estimation settings.
+	GetCostEstimation(ctx context.Context) (*AdminCostEstimationSetting, error)
+
+	// UpdateCostEstimation updates the cost estimation settings.
+	UpdateCostEstimation(ctx context.Context, options AdminCostEstimationSettingOptions) (*AdminCostEstimationSetting, error)
+
+	// GetSAML returns the SAML settings.
+	GetSAML(ctx context.Context) (*AdminSAMLSetting, error)
+
+	// UpdateSAML updates the SAML settings.
+	UpdateSAML(ctx context.Context, options AdminSAMLSettingsUpdateOptions) (*AdminSAMLSetting, error)
+
+	// RevokeSAMLIdpCert revokes the older IdP certificate when the new IdP
+	// certificate is known to be functioning correctly.
+	RevokeSAMLIdpCert(ctx context.Context) (*AdminSAMLSetting, error)
+
+	// GetSMTP returns the SMTP settings.
+	GetSMTP(ctx context.Context) (*AdminSMTPSetting, error)
+
+	// UpdateSMTP updates SMTP settings.
+	UpdateSMTP(ctx context.Context, options AdminSMTPSettingsUpdateOptions) (*AdminSMTPSetting, error)
+
+	// GetTwilio returns the Twilio settings.
+	GetTwilio(ctx context.Context) (*AdminTwilioSetting, error)
+
+	// UpdateTwilio updates Twilio settings.
+	UpdateTwilio(ctx context.Context, options AdminTwilioSettingsUpdateOptions) (*AdminTwilioSetting, error)
+
+	// VerifyTwilio verifies Twilio settings.
+	VerifyTwilio(ctx context.Context, options AdminTwilioSettingsVerifyOptions) error
+
+	// GetCustomization returns the customization settings.
+	GetCustomization(ctx context.Context) (*AdminCustomizationSetting, error)
+
+	// UpdateCustomization updates the customization settings.
+	UpdateCustomization(ctx context.Context, options AdminCustomizationSettingsUpdateOptions) (*AdminCustomizationSetting, error)
+}
```

Comment:
Random thought, take or leave it: Does it make sense, especially as more and more stuff will undoubtedly be added to this, to further differentiate areas by one more layer of structing? `Client.Admin.Settings.SAML.Read`, `Client.Admin.Settings.SAML.RevokeIdpCert`, etc? I'm torn really and could go either way, but thought I'd bring it up to ponder about at least.

Reply:
One thing lending to this organization is something I didn't notice before, which is that the "Settings API" is actually a collection of a bunch of different ones 🤔 `admin/cost-estimation-settings`, `admin/saml-settings`, etc.

Reply:
Thats a very fair point. I think thats a better approach. I'll update